lib/src/main/java/graphql/nadel/engine/NadelDeferSupport.kt

```diff
@@ -0,0 +1,167 @@
+package graphql.nadel.engine
+
+import graphql.incremental.DelayedIncrementalPartialResult
+import graphql.nadel.engine.NadelDeferSupport.OutstandingJobCounter.OutstandingJobHandle
+import graphql.nadel.engine.util.copy
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+class NadelDeferSupport internal constructor(
+    private val delayedResultsChannel: Channel<DelayedIncrementalPartialResult> = Channel(UNLIMITED),
```

Comment:
What does `UNLIMITED` mean here? Is it the maximum size of the buffer? Should we adopt a reasonable limit instead?

Reply:
Yeah it's the size of the buffer, we can drop it down.

Reply:
Dropped it down to 100 which should be much more than we need, and yet still a sane limit.